ciao-controller/api/api.go

```diff
@@ -440,12 +440,29 @@ func addWorkload(c *Context, w http.ResponseWriter, r *http.Request) (Response,
 		return errorResponse(err), err
 	}
 
+	// we allow admin to create public workloads for any tenant. However,
+	// users scoped to a particular tenant may only create workloads
+	// for their own tenant.
+	vars := mux.Vars(r)
+	tenantID, ok := vars["tenant"]
+	if ok {
+		req.TenantID = tenantID
+	} else {
+		req.TenantID = "public"
```

Comment:
the identity code we use checks that your token is scoped to the tenantID - so that prevents normal users from messing with "public". I guess if you had admin rights you might be able to come in using tenantID "public" on apis where we haven't added the mux regex matching for the uuid4 tenantID yet (the openstack ones I believe) - but I think if you have admin rights you can pretty much do whatever you want to the entire cluster anyway.